spring-rabbit/src/main/java/org/springframework/amqp/rabbit/config/SimpleRabbitListenerContainerFactory.java

```diff
@@ -134,43 +136,37 @@ protected SimpleMessageListenerContainer createContainerInstance() {
 		return new SimpleMessageListenerContainer();
 	}
 
-	@Override // NOSONAR complexity = mostly null checks
+	@Override
 	protected void initializeContainer(SimpleMessageListenerContainer instance, RabbitListenerEndpoint endpoint) {
 		super.initializeContainer(instance, endpoint);
 
-		if (this.txSize != null) {
-			instance.setTxSize(this.txSize);
-		}
+		setIfNotNull(this.txSize, instance::setTxSize);
 		String concurrency = null;
 		if (endpoint != null) {
 			concurrency = endpoint.getConcurrency();
-			if (concurrency != null) {
-				instance.setConcurrency(concurrency);
-			}
-		}
-		if (concurrency == null && this.concurrentConsumers != null) {
-			instance.setConcurrentConsumers(this.concurrentConsumers);
-		}
-		if ((concurrency == null || !(concurrency.contains("-"))) && this.maxConcurrentConsumers != null) {
-			instance.setMaxConcurrentConsumers(this.maxConcurrentConsumers);
-		}
-		if (this.startConsumerMinInterval != null) {
-			instance.setStartConsumerMinInterval(this.startConsumerMinInterval);
-		}
-		if (this.stopConsumerMinInterval != null) {
-			instance.setStopConsumerMinInterval(this.stopConsumerMinInterval);
+			setIfNotNull(concurrency, instance::setConcurrency);
 		}
-		if (this.consecutiveActiveTrigger != null) {
-			instance.setConsecutiveActiveTrigger(this.consecutiveActiveTrigger);
-		}
-		if (this.consecutiveIdleTrigger != null) {
-			instance.setConsecutiveIdleTrigger(this.consecutiveIdleTrigger);
-		}
-		if (this.receiveTimeout != null) {
-			instance.setReceiveTimeout(this.receiveTimeout);
+		setIf(concurrency == null && this.concurrentConsumers != null, this.concurrentConsumers,
+				instance::setConcurrentConsumers);
+		setIf((concurrency == null || !(concurrency.contains("-"))) && this.maxConcurrentConsumers != null,
+				this.maxConcurrentConsumers, instance::setMaxConcurrentConsumers);
+		setIfNotNull(this.startConsumerMinInterval, instance::setStartConsumerMinInterval);
+		setIfNotNull(this.stopConsumerMinInterval, instance::setStopConsumerMinInterval);
+		setIfNotNull(this.consecutiveActiveTrigger, instance::setConsecutiveActiveTrigger);
+		setIfNotNull(this.consecutiveIdleTrigger, instance::setConsecutiveIdleTrigger);
+		setIfNotNull(this.receiveTimeout, instance::setReceiveTimeout);
+		setIfNotNull(this.deBatchingEnabled, instance::setDeBatchingEnabled);
+	}
+
+	private <T> void setIf(boolean predicate, T value, Consumer<T> instance) {
+		if (predicate) {
+			instance.accept(value);
 		}
-		if (this.deBatchingEnabled != null) {
-			instance.setDeBatchingEnabled(this.deBatchingEnabled);
+	}
+
+	private <T> void setIfNotNull(T value, Consumer<T> instance) {
```

Comment:
Sounds like a public utility which can be applied in many other places. However its name sounds better like `acceptIfNotNull()` 😄

Reply:
Agreed - maybe `JavaUtils`, `CodeUtils` ?

`configureIfNotNull` ?

Reply:
In Spring Boot they have something similar in the `PropertyMapper`.
We may think about something similar and evolve eventually.

From the current perspective an implementation is not really about *configure*, but just a conditional call for the provided `Consumer`, but I can live with whatever you think is better for us